java/src/main/java/com/codurance/training/tasks/TaskList.java

```diff
@@ -143,7 +134,84 @@ private void error(String command) {
         out.println();
     }
 
-    private long nextId() {
-        return ++lastId;
+    private void setDeadLine(String command){
+        String[] subcommandRest = command.split(" ", 2);
+        String taskId = subcommandRest[0];
+        String deadLine = subcommandRest[1];
+
+        for (Map.Entry<String, List<Task>> project : tasks.entrySet()) {
+            for (Task task : project.getValue()) {
+                if (task.getId().equals(taskId)) {
+                    task.setDeadline(LocalDate.parse(deadLine));
+                    return;
+                }
+            }
+        }
+        out.printf("Could not find a task with an ID of %s.%n", taskId);
+    }
+
+    private void showTodayTasks(){
+        LocalDate currentDate = LocalDate.now();
+
+        for (Map.Entry<String, List<Task>> project : tasks.entrySet()) {
+            out.println(project.getKey());
+            for (Task task : project.getValue()) {
+                LocalDate deadline = task.getDeadline();
+                if (deadline != null && deadline.equals(currentDate)) {
+                    out.printf("    [%c] %s: %s : %s%n", (task.isDone() ? 'x' : ' '), task.getId(),
+                            task.getDescription(), deadline);
+                }
+            }
+            out.println();
+        }
+    }
+
+    public void delete(String strId) {
+        for (List<Task> projectTasks : tasks.values()) {
+            for (Task task : projectTasks) {
+                if (task.getId().equals(strId)) {
+                    projectTasks.remove(task);
+                    return;
+                }
+            }
+        }
+        out.printf("Could not find a task with an ID of %s.%n", strId);
+    }
+
+    public void viewTasks(String commandLine){
+        String[] subcommandRest = commandLine.split(" ", 3);
+        String subcommand = subcommandRest[1];
+        if(subcommand.equals("date")){
+            viewTasksBydate(subcommandRest[2]);
+        }else if(subcommand.equals("project")){
+            taskView.showTasks(tasks);
+        }
+        else {
+            viewTasksByDeadLine();
+        }
+    }
+
+    private void viewTasksBydate(String date) {
+        for (Map.Entry<String, List<Task>> project : tasks.entrySet()) {
+            for (Task task : project.getValue()) {
+                if (task.getDeadline() != null && task.getDeadline().toString().equals(date)) {
+                    out.printf("    [%c] %s: %s%n", (task.isDone() ? 'x' : ' '), task.getId(), task.getDescription());
+                }
+            }
+            out.println();
+        }
+    }
+
+    public void viewTasksByDeadLine(){
+        List<Task> allTasks = new ArrayList<>();
+        for (List<Task> tasks : this.tasks.values()) {
+            allTasks.addAll(tasks);
+        }
```

Comment:
same here... for viewTasksByDeadline viewTasksBydate etc... u can create one interface which will be implemented by different classes